### PR TITLE
Support projectV2 for changelog tooling [skip ci]

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

To support generate changelog based on project V2 https://github.com/orgs/NVIDIA/projects/4
1. New project V2 API
2. keep compatible of project V1
3. improve generation speed

Updated new changelog to latest,
```
GITHUB_TOKEN=<PAT> scripts/generate-changelog --releases=22.02,22.04,22.06,22.08,22.10,22.12
```

### CODE REVIEW

**Comment by mattahrens** [comment]
Just to confirm, can we delete the repo-specific projects once this is merged?  https://github.com/NVIDIA/spark-rapids/projects?type=classic

**Comment by pxLi** [comment]
> 

What does `delete` mean? 
if `closed`, we should be always fine, 
if delete, it will fail to fetch the deleted when try to re-generate changelog for elder release like 22.10

I would prefer close and do not create new classic/V1 project but not delete.

**Comment by mattahrens** [comment]
Yes, that makes sense in only closing existing projects. So then starting with 22.12 (meaning anytime), we will no longer assign issues to Projects by release version; they will be assigned only to the spark-rapids global project in Github.

File: scripts/generate-changelog [hunk 1]
```diff
@@ -39,25 +39,26 @@ e.g.
 Dependencies:
     - requests
 
-Github personal access token: https://github.com/settings/tokens, and make you have `repo` scope selected
+Github personal access token: https://github.com/settings/tokens, and make you have `repo` and `project` scope selected
 
 Usage:
     cd spark-rapids/
 
     # generate changelog for releases 22.02 to 22.10
```
Comment (sameerz) [inline scripts/generate-changelog:47]:
```suggestion
    # generate changelog for releases 22.02 to 22.12
```

File: scripts/generate-changelog [hunk 2]
```diff
@@ -39,25 +39,26 @@ e.g.
...
     scripts/generate-changelog --token=<GITHUB_PERSONAL_ACCESS_TOKEN> \
-    --releases=22.02,22.04,22.06,22.08,22.10
+    --releases=22.02,22.04,22.06,22.08,22.10,22.12
 
     # generate changelog for releases 22.02 to 22.10 to /tmp/CHANGELOG.md
```
Comment (sameerz) [inline scripts/generate-changelog:51]:
```suggestion
    # generate changelog for releases 22.02 to 22.12 to /tmp/CHANGELOG.md
```

**Review by sameerz — COMMENTED** [review]
The command 

`scripts/generate-changelog --releases 22.12 --path=/tmp/CHANGELOG.md`

Did not generate any output for me, nor when I tried with release 22.10.  (Github token was specified in environment variable). 

I got the following output:

```
Generating changelog ...
Processing pull requests ...
'data'
```

What am I doing wrong? :-)

**Comment by pxLi** [comment]
> 

Not quite sure about your ENV,
```
> GITHUB_TOKEN=<token> scripts/generate-changelog --releases 22.10 --path=/tmp/CHANGELOG.md
Generating changelog ...
Processing pull requests ...
Processing issues ...
Done.
```

Can you double check that TOKEN you are using w/ correct scopes? since new change requires project scope

**Comment by sameerz** [comment]
> Can you double check that TOKEN you are using w/ correct scopes? since new change requires project scope

That was the issue, my token did not have the right scopes.

**Review by sameerz — APPROVED** [review]

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
Going to merge this one as initial changelog update of project V2, 
we will update changelog later when code freeze~

**Pull request merged**